sim/mage/heating_up.go

```diff
@@ -39,21 +39,17 @@ func (mage *Mage) registerHeatingUp() {
 
 func (mage *Mage) HeatingUpSpellHandler(sim *core.Simulation, spell *core.Spell, result *core.SpellResult, callback func()) {
 	if spell.TravelTime() > time.Duration(FireSpellMaxTimeUntilResult) {
-		pa := sim.GetConsumedPendingActionFromPool()
-		pa.NextActionAt = sim.CurrentTime + time.Duration(FireSpellMaxTimeUntilResult)
-
-		pa.OnAction = func(sim *core.Simulation) {
+		spell.RegisterTravelTimeCallback(sim, time.Duration(FireSpellMaxTimeUntilResult), func(sim *core.Simulation) {
 			callback()
 			mage.HandleHeatingUp(sim, spell, result)
-		}
-
-		sim.AddPendingAction(pa)
+		})
 	} else {
 		spell.WaitTravelTime(sim, func(sim *core.Simulation) {
 			callback()
 			mage.HandleHeatingUp(sim, spell, result)
 		})
 	}
```

Comment:
This can now be simplified to just:
```
spell.RegisterTravelTimeCallback(sim, min(spell.TravelTime(), FireSpellMaxTimeUntilResult), func(sim *core.Simulation) {
    callback()
    mage.HandleHeatingUp(sim, spell, result)
})
```

Reply:
oo nice yeah, sec